user-guide/Standard_Apps/IDP/Advanced_configuration/Implementing the Connectivitivty Discovery script.md

```diff
@@ -0,0 +1,102 @@
+
+> [!TIP]
+> An example script *IDP_Example_Custom_ConnectivityDiscovery* is available in the Automation module after IDP has been installed. You can duplicate this script to use it as a starting point.
+
+## Unresolved connections
+
+The Connectivity Discovery script needs to call certain C# methods to inform IDP of the neighborships discovered on the element. However, the information the element can provide will typically not be enough to provision DCF connections. That is why we call these **unresolved connections**: the detected neighborship does not have enough information for DCF connections to be provisioned.
+
+There is only enough information when a connectivity discovery on another element leads to another unresolved connection that can be combined with the first. DataMiner IDP will automatically combine two unresolved connections into a **discovered connection**. A discovered connection contains enough information to be able to provision a DCF connection for it.
+
+## Combining unresolved connections into discovered connections
+
+A DCF connection exists between two DCF interfaces of two elements. As such, to provision a DCF connection, the following things need to be known:
+
+- The source DMA and element ID
+- The source DCF interface ID
+- The destination DMA and element ID
+- The destination DCF interface ID
+
+At the same time, the custom script is linked to the CI Type and should only use the information available in the element itself.
+
+A network protocol that allows the equipment to discover neighboring network equipment (like LLDP, CDP, etc.) will have:
+
+- A unique way to identify equipment. The equipment will be given an identifier that no other equipment is using. We refer to this as the **Local Chassis ID**.
+- A unique way to identify interfaces on the equipment. This can be different from the typical interface ID. We refer to this as a **Local Port ID**.
+- The ability to see on which interfaces neighbors are detected. The neighbors are detected with their unique identifiers. From the perspective of local equipment, these are **Remote Chassis IDs**.
+
+If the equipment runs such a network protocol, the connector will have or will need to be extended with the information above. The custom script can then provide IDP with the following information regarding neighbors of the equipment:
+
+- The DataMiner and element ID of the element.
+- The unique identifier (i.e. the Local Chassis ID) of the equipment.
+- For each neighbor of the equipment:
+
+  - The DCF interface ID on which the neighbor is detected.
+  - The local port ID on which the neighbor is connected.
+  - The remote chassis ID of the neighbor.
+

```

Comment:
- the remote port ID (this is port ID which uniquely identifies the interface on the neighbor)